extensions/ql-vscode/src/vscode-tests/no-workspace/remote-query-history.test.ts

```diff
@@ -0,0 +1,252 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as sinon from 'sinon';
+import * as chai from 'chai';
+import 'mocha';
+import 'sinon-chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import { ExtensionContext, Uri, window, workspace } from 'vscode';
+import { QueryHistoryConfig } from '../../config';
+import { DatabaseManager } from '../../databases';
+import { tmpDir } from '../../helpers';
+import { QueryHistoryManager } from '../../query-history';
+import { QueryServerClient } from '../../queryserver-client';
+import { DisposableBucket } from '../../pure/disposable-object';
+import { Credentials } from '../../authentication';
+import { AnalysesResultsManager } from '../../remote-queries/analyses-results-manager';
+import { RemoteQueryResult } from '../../remote-queries/shared/remote-query-result';
+
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+/**
+ * Tests for remote queries and how they interact with the query history manager.
+ */
+
+describe('Remote queries and query history manager', function() {
+
+  const EXTENSION_PATH = path.join(__dirname, '../../../');
+  const STORAGE_DIR = path.join(tmpDir.name, 'remote-queries');
+  const asyncNoop = async () => { /** noop */ };
+
+  let sandbox: sinon.SinonSandbox;
+  let qhm: QueryHistoryManager;
+  let rawQueryHistory: any;
+  let remoteQueryResult0: RemoteQueryResult;
+  // let remoteQueryResult1: RemoteQueryResult;
+  let disposables: DisposableBucket;
+  let showTextDocumentSpy: sinon.SinonSpy;
+  let openTextDocumentSpy: sinon.SinonSpy;
+
+  beforeEach(() => {
+    copyHistoryState();
+  });
+
+  afterEach(() => {
+    deleteHistoryState();
+  });
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+    disposables = new DisposableBucket();
+    rawQueryHistory = fs.readJSONSync(path.join(STORAGE_DIR, 'workspace-query-history.json'));
+    remoteQueryResult0 = fs.readJSONSync(path.join(STORAGE_DIR, 'queries', rawQueryHistory[0].queryId, 'query-result.json'));
+    // remoteQueryResult1 = fs.readJSONSync(path.join(STORAGE_DIR, 'queries', rawQueryHistory[1].queryId, 'query-result.json'));
```

Comment:
Should this be removed?

Reply:
I'll be writing more tests and will likely need this variable.

Reply:
Shall we add it then? Happy either way.